### PR TITLE
DDF-2769 Fixed condition to determine whether to run SchematronValidation in SchematronValidationService

### PR DESCRIPTION
**Pull request by emanns95**
#### What does this PR do?
Condition causes all schematron validators to run on any metadata. Fixes that.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@brendan-hofmann 
@ahoffer 

#### Select at least one member from relevant component team(s) from below (at least one component team member needs to approve the PR).
[Build](https://github.com/orgs/codice/teams/build)

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@andrewkfiedler
@coyotesqrl
@stustison

#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Change Log Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java [hunk 1]
```diff
@@ -310,8 +310,8 @@ private MetacardValidationReport generateReport(Metacard metacard)
         MetacardValidationReportImpl report = new MetacardValidationReportImpl();
         Set<String> attributes = ImmutableSet.of("metadata");
         String metadata = metacard.getMetadata();
-        boolean canBeValidated = StringUtils.isNotEmpty(metadata) || namespace == null
-                || namespace.equals(XMLUtils.getRootNamespace(metadata));
+        boolean canBeValidated = StringUtils.isNotEmpty(metadata) && namespace == null
```
Comment (shaundmorris) [inline catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java]: Should there have been a unit test that caught this issue?

Comment (coyotesqrl) [inline catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java]: As was pointed out out-of-band, should be `namespace != null`.

Comment (emanns95) [inline catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java]: Probably don't have any trying to run the wrong validators on a metacard

File: catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java [hunk 2]
```diff
@@ -310,8 +310,8 @@ private MetacardValidationReport generateReport(Metacard metacard)
         MetacardValidationReportImpl report = new MetacardValidationReportImpl();
         Set<String> attributes = ImmutableSet.of("metadata");
         String metadata = metacard.getMetadata();
-        boolean canBeValidated = StringUtils.isNotEmpty(metadata) || namespace == null
-                || namespace.equals(XMLUtils.getRootNamespace(metadata));
+        boolean canBeValidated = StringUtils.isNotEmpty(metadata) && namespace != null
```
Comment (coyotesqrl) [inline catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java]: @shaundmorris comment about a unit test stands. Even if it is not submitted today, a ticket should be opened to create one.

Comment (shaundmorris) [inline catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java]: agree, @emanns95 can you please create the ticket and link it?

Comment (emanns95) [inline catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java]: https://codice.atlassian.net/browse/DDF-2784

**Pull request merged**